### DISCUSSION TITLE
Water fountains can have bottle fillers too

### DISCUSSION THREAD

**Discussion by endolith** [streetcomplete/StreetComplete#4632] [Feedback]
The quest for water fountains gives multiple options, but doesn't allow to combine them.  It should behave more like a checkbox  than a radio button, allowing you to say that a fountain has both a bottle filler and a drinking fountain, for instance.

![Screenshot_20221114-115641_StreetComplete](https://user-images.githubusercontent.com/58611/201722060-26db974f-78b0-4b3f-9391-e861218d3c93.jpg)

![Water_Fountain_and_Bottle_Filler_in_Flight_of_Passage_Queue_(34585701671) 1](https://user-images.githubusercontent.com/58611/201721843-bcb4f9a2-f0db-40c1-b05b-01cef7f55749.jpg)

**Comment by mnalis** [comment]
You seem to have missed the word "only" in "Refilling bottles **only**" answer. So if there are _other ways_ to consume water from that amenity, than that one is **not** the correct answer, and you should choose another answer which best describes the amenity.

Because, you can fill bottles also at "water fountain emitting small jet of water", "water tap", "hand pump", "other water fountain" etc....

And yeah, there will always be special cases which are similar, but not exactly as one in the pictures... Just choose the one which matches the amenity **the most**. If still unsure, choose "Other water fountain" which is supposed to cover all special cases which are not one of the other displayed answers. 

So, if I'm interpreting your picture correctly, it is _"Water fountain emitting small jet of water"_.

**Reply by endolith** [reply]
No I didn't miss the word. That's the whole point of my feedback. Some have a bottle filler, some do not. Some have a water fountain, some do not. Some have a spigot, some do not.

**Reply by mnalis** [reply]
Then I am missing the point. :man_shrugging: Could you try to explain using other words what exactly is the problem / what are you trying to accomplish and why?

If I am understanding correctly, you seem to want (for some reason) to tag **multiple** fountain types **at the same time**?

But how **exactly** do you envision you expect the result to be **tagged**? For example, **what tags do you expect to be set on OSM element when all 4 of the following pictures are selected**:
- "water fountain emitting small jet of water"
- "water tap"
- "hand pump"
- "refilling bottles"

(To see what is *currently* tagged by some answer, choose an answer, and then press Undo button in lower left corner.)

Because I myself cannot understand **what would you want to be tagged there**, as they use conflicting tags. (e.g. `fountain=*` can have only one value; if you tag [fountain=bottle_refill](https://wiki.openstreetmap.org/wiki/Key:fountain)  you cannot at the same time tag [fountain=bubbler](https://wiki.openstreetmap.org/wiki/Tag:fountain%3Dbubbler). Same thing with [man_made=water_tap](https://wiki.openstreetmap.org/wiki/Tag:man_made%3Dwater_tap) vs. [man_made=water_well](https://wiki.openstreetmap.org/wiki/Tag:man_made%3Dwater_well), etc.)

Or do you want to suggest new OSM tags, which could be made in such way so they can marked multiple times on the same element? StreetComplete github is wrong place do that, see OSM [Tagging mailing list](https://wiki.openstreetmap.org/wiki/Tagging_mailing_list) for that (as StreetComplete cannot _invent_ new tags, but only use existing ones already discussed, used and documented at https://wiki.openstreetmap.org/ )

Or is it something else? As I am unsure of how you would want to map (and later use) such data, I can only try to guess, sorry.

Also, see this (related to bottle filling) ticket: https://github.com/streetcomplete/StreetComplete/issues/4110

**Reply by endolith** [reply]
Yes, we need the ability to tag multiple fountain types at the same time, for the same node. The tag wiki describes at least two such combinations:

> If it is part of the same installation as a fountain you can drink directly from, use fountain=[bubbler](https://wiki.openstreetmap.org/wiki/Tag:fountain%3Dbubbler) and [bottle](https://wiki.openstreetmap.org/wiki/Key:bottle)=[yes](https://wiki.openstreetmap.org/w/index.php?title=Tag:bottle%3Dyes&action=edit&redlink=1) in preference to creating multiple points.

Likewise:

> Drinking fountain which has a tap would be tagged with both fountain=drinking and man_made=water_tap 

Or with both [fountain=bubbler](https://wiki.openstreetmap.org/wiki/Tag:fountain%3Dbubbler) and [man_made=water_tap](https://wiki.openstreetmap.org/wiki/Tag:man_made%3Dwater_tap)

> For example, what tags do you expect to be set on OSM element when all 4 of the following pictures are selected:
> 
> -    "water fountain emitting small jet of water"
> -    "water tap"
> -    "hand pump"
> -    "refilling bottles"

I don't know why it would have a hand pump *and* a water tap.  Without that, it should be tagged with [fountain=bubbler](https://wiki.openstreetmap.org/wiki/Tag:fountain%3Dbubbler) and [bottle](https://wiki.openstreetmap.org/wiki/Key:bottle)=[yes](https://wiki.openstreetmap.org/w/index.php?title=Tag:bottle%3Dyes&action=edit&redlink=1) and [man_made=water_tap](https://wiki.openstreetmap.org/wiki/Tag:man_made%3Dwater_tap) according to the tag wiki.

But yes, I don't know the best way to tag other combinations, like a bubbler and a dog bowl fountain.

> (e.g. fountain=* can have only one value; if you tag [fountain=bottle_refill](https://wiki.openstreetmap.org/wiki/Key:fountain) you cannot at the same time tag [fountain=bubbler](https://wiki.openstreetmap.org/wiki/Tag:fountain%3Dbubbler).

Maybe [semicolons](https://wiki.openstreetmap.org/wiki/Semi-colon_value_separator)?

https://taginfo.openstreetmap.org/tags/fountain=bottle_refill%3Bbubbler

> (as StreetComplete cannot invent new tags, but only use existing ones already discussed, used and documented at [wiki.openstreetmap.org](https://wiki.openstreetmap.org/) )

Why can't StreetComplete invent new tags?
https://wiki.openstreetmap.org/wiki/Any_tags_you_like

**Reply by peternewman** [reply]
I'd agree with @endolith , I think there will be sufficient appearances of bubble with a bottle refill station (and the fact it's mentioned explicitly on the wiki) that we should add a separate option for that which tags `fountain=bubbler` and `bottle=yes` as recommended on the wiki. Google has lots of examples of different types:
https://www.google.com/search?q=water+fountain+bottle+refill&tbm=isch

By NOT tagging bottle=no in the other cases, we avoid the vagueness of this tagging scheme (presumably you could refill a bottle slowly from a bubbler only fountain), but do provide some benefit to consumers of that tag.

Likewise I think avoiding having all the other combinations continues to avoid the complicated edge cases.

Incidentally we're not tagging `bottle=yes` when we tag `fountain=bottle_refill` despite it being requested on https://wiki.openstreetmap.org/wiki/Key:fountain which seems to have the same positive side effect.

**Reply by mnalis** [reply]
> Why can't StreetComplete invent new tags?

see  https://wiki.openstreetmap.org/wiki/Any_tags_you_like#When_to_create_a_proposal especially from `Your feature is of general interest` and below. Also, it is a good policy IMHO. Note that StreetComplete has a lot of users - any tag it might invent would be massively used, and if not accepted by community, would cause massive amount of complaints. Even quite accepted and documented features like `feature=no` have repeatedly drawn enough negative attention that a [FAQ](https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ#Why_does_StreetComplete_often_tag_the_absence_of_features?) entry had to be written explaining the SC position.

As for the multichoice - it is a bad choice (no pun intended!) if **not all** pictures can be combined in **all** ways (as they can't). Also, semicolon tagging is often not supported by data consumers (speaking of data consumers; where/how do you envision collected data proposed here to be used? Do you already have some app which is able to use it?) 
If only some combinations should be accepted, then multi-choice should not be used, but instead single-choice should be used, and combo-answers should have their own pictures. But then one sacrifices simplicity and provoke bad tagging (e.g. users might likely click on first picture that looks correct, even if there are other pictures which describe situation on the ground more precisely).

I personally would probably mark unknown combinations as `other water fountain` (which it is), and not worry about details. Or if I felt very strongly about usefulness, I'd select `other answers` / `leave a note` and then mapped it as specifically as I wanted when I got home (or even open Vespucci/Everydoor on the go and edited directly there). Yet I guess I would feel strongly enough only very rarely with this particular quest (if ever). But that is just me, YMMV.

**Reply by matkoniecz** [reply]
> Why can't StreetComplete invent new tags?

It is more of self-limitation done for practical and political reasons.

There can be still new tags invented as during StreetComplete development it was discovered that they are needed, but discussing them should not take place on SC issue tracker. But rather in place where wider OSM community is present (community.openstreetmap.org or tagging mailing list for example).

**Reply by codeinabox** [reply]
> But yes, I don't know the best way to tag other combinations, like a bubbler and a dog bowl fountain.

Could we have a quest if `amenity=drinking_water` has a dog bowl?

**Reply by mnalis** [reply]
> Could we have a quest if amenity=drinking_water has a dog bowl?

We could, if it checks all the boxes? Click [here to create new issue](https://github.com/streetcomplete/StreetComplete/issues/new/choose) and choose `Quest suggestion` from the list. It will fill the template which contains instructions and checkboxes, which you check or uncheck and answer the questions.

**Reply by mnalis** [reply]
> Could we have a quest if amenity=drinking_water has a dog bowl?

- ref: https://github.com/streetcomplete/StreetComplete/issues/6149

**Comment by rhhsm** [comment]
This quest has me confused too, but for a different reason. Most of the drinking water sources I like to tag are rural/mountain area sources that are often springs, but don't look like any of the pictures of the quest. They typically look like this 
![IMG_20220605_095552660_HDR](https://user-images.githubusercontent.com/69100427/202908033-4b74de40-1258-4609-92cf-a3e56333ca47.jpg)
They're called "cesme" and similar words in South_eastern Europe and the Middle East. Which option should I choose for answering this quest?

**Reply by peternewman** [reply]
> This quest has me confused too, but for a different reason. Most of the drinking water sources I like to tag are rural/mountain area sources that are often springs, but don't look like any of the pictures of the quest.

The wiki ( https://wiki.openstreetmap.org/wiki/Tag:amenity%3Ddrinking_water ) says:

> [natural](https://wiki.openstreetmap.org/wiki/Key:natural)=[spring](https://wiki.openstreetmap.org/wiki/Tag:natural%3Dspring) indicates water flowing naturally out of the ground. It may have an [amenity](https://wiki.openstreetmap.org/wiki/Key:amenity)=[fountain](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dfountain) built around it, for example see [Roman Bath](http://www.romanbaths.co.uk/) in [Bath](https://www.openstreetmap.org/?lat=51.38078&lon=-2.35967&zoom=17&layers=B0T), U.K.

Which seems to answer the how to tag it bit.

> Which option should I choose for answering this quest?

So I'd guess from that and the code picking spring would be appropriate.

I think where it gets challenging is that as an outsider I wouldn't be able to tell whether that was fed from a spring or a water pipe (aside from the lack of presence of a tap. I don't know how you'd survey that bit unless it was signed?

**Reply by rhhsm** [reply]
Maybe the image for "spring" could be replaced by one which shows water in a more easily drinkable form (I find it hard to imagine how to drink water from the spring shown in the image, and wouldn't think of it primarily as a source of drinking water). The image I was tempted to choose is "other water fountain".
What tags are added to the node when choosing each of the options?

**Reply by peternewman** [reply]
> Maybe the image for "spring" could be replaced by one which shows water in a more easily drinkable form (I find it hard to imagine how to drink water from the spring shown in the image, and wouldn't think of it primarily as a source of drinking water).

You could fill a bottle from it for example. I think it's more in the traditional map sense of the spring is the source of the water and would be marked on a map (compared to part way down a mountain stream where you could also fill a bottle).

> The image I was tempted to choose is "other water fountain".

As I say, I don't think that would be unfair given what you're seeing.

> What tags are added to the node when choosing each of the options?

You can see the tagging here:
https://github.com/streetcomplete/StreetComplete/blob/58b054a4192cbdaa98a63c6434af37dc45020047/app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water_type/DrinkingWaterType.kt#L21-L43

**Reply by matkoniecz** [reply]
>  What tags are added to the node when choosing each of the options?

You can see also them in undo menu. You can enable airplane mode for the entire phone (or manual upload mode in sc) and experiment with answers and undo test edits - and while undoing edits you can see tags applied, even with links to wiki.
